### PR TITLE
Add gauge visualization

### PR DESCRIPTION
**Pull request by IdoPeles**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description
This if my first time contributing to an open source project , please excuse my mistakes and let me know what should I do next if needed

This feature is a simple arc displaying 3 colors and a value used to describe status
value should be passed in column called "value" result set should return one row

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
gauge

### CODE REVIEW

File: client/app/visualizations/gauge/index.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  },
+  controller($scope, $element) {
+    const container = $element[0].querySelector('.power-gauge');
+
+    const update = () => {
+      if (this.data && isDataValid(this.data)) {
+        // do the render logic.
+        angular.element(container).empty();
+
+        createGauge(container, this.data.rows, this.options).render(this.data.rows[0].value);
```
Comment (kravets-levko) [inline client/app/visualizations/gauge/index.js]: 1. When you create something using some library - you have to destroy it using the same library. In this case probably everything will be fine (but I'm not 100% sure tho), but you create svg with `d3` - so you shouldn't use Angular's functions to destroy it.
2. You return an object from `createGauge` with a lot of functions, but later use only `render`. I think, it's better to rename it to `renderGauge`, pass all needed arguments to it and do not return anything from it at all.

**Comment by kravets-levko** [comment]
I realized that Plotly (library we use for a lot of other charts) already have a gauge chart: https://plot.ly/javascript/gauge-charts/ 

WDYT about using it instead of your custom implementation?

**Comment by deecay** [comment]
> I realized that Plotly (library we use for a lot of other charts) already have a gauge chart: 
I was just going to mention that. 

[Previous discussion](https://discuss.redash.io/t/funnel-chart-by-plotly/4019). See Indicator examples.  
Is this worth the increased dependency?

File: client/app/visualizations/gauge/Editor.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  const updateOptions = updates => onOptionsChange({
+    ...options,
+    ...updates,
+  });
+
+  return (
+    <div className="form-group">
+      <label className="d-flex flex-column m-b-10" htmlFor="pivot-show-controls">
+        <span className="m-r-10">Ok Range</span>
+        <input className="form-control" type="text" value={options.okRange} onChange={event => updateOptions({ okRange: event.target.value })} />
```
[the remark this answers is not in the document]
Comment (IdoPeles) [reply client/app/visualizations/gauge/Editor.jsx]: image
changing className to class causes errors, see attachment

regarding the colorPallet I would be happy to replace but I did not understand how erectly to use it , can you explain how it is used

File: client/app/visualizations/gauge/Editor.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+export default function Editor({ options, onOptionsChange }) {
+  const updateOptions = updates => onOptionsChange({
+    ...options,
+    ...updates,
+  });
+
+  return (
+    <div className="form-group">
+      <label className="d-flex flex-column m-b-10" htmlFor="pivot-show-controls">
```
[the remark this answers is not in the document]
Comment (IdoPeles) [reply client/app/visualizations/gauge/Editor.jsx]: done

File: client/app/visualizations/gauge/gauge.less [hunk 4]
```diff
@@ -0,0 +1,15 @@
+.power-gauge g.arc {
+  fill: steelblue;
+}
+
+.power-gauge g.pointer {
```
[the remark this answers is not in the document]
Comment (IdoPeles) [reply client/app/visualizations/gauge/gauge.less]: done

File: client/app/visualizations/gauge/index.js [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+      [-(config.pointerWidth / 2), 0],
+      [0, config.pointerTailLength],
+      [config.pointerWidth / 2, 0]];
+    const pointerLine = d3.svg.line().interpolate('monotone');
+    const pg = svg.append('g').data([lineData])
+      .attr('class', 'pointer')
+      .attr('transform', centerTx);
+
+    pointer = pg.append('path')
+      .attr('d', pointerLine/* function(d) { return pointerLine(d) +'Z';} */)
```
[the remark this answers is not in the document]
Comment (IdoPeles) [reply client/app/visualizations/gauge/index.js]: done

File: client/app/visualizations/gauge/index.js [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+  return that;
+}
+
+function isDataValid(data) {
+  // data should contain column named 'value', otherwise no reason to render anything at all
+  return _.find(data.columns, c => c.name === 'value');
+}
+
+const GaugeRenderer = {
```
[the remark this answers is not in the document]
Comment (IdoPeles) [reply client/app/visualizations/gauge/index.js]: I need assistance , no experience in react , give me a reference I'll study it

File: client/app/visualizations/gauge/index.js (hunk 1 above)
Comment (IdoPeles) [reply client/app/visualizations/gauge/index.js]: encapsulated the internal methods 
regarding destroing the d3 object I used the code from "sankey" and I don't see there something related , maybe you can reference me to relevant directions

**Comment by kravets-levko** [comment]
Hey @IdoPeles,

I converted renderer component to React as well - you can pull latest changes I see how I did it. But your repo is quite out of date with ours, for example [resizeObserver](https://github.com/getredash/redash/blob/master/client/app/services/resizeObserver.js) service is missing (but it's needed to replace a `resize-event` directive). So I kindly ask you to catch up with our repo, and then we can continue with this visualization.

File: client/app/visualizations/gauge/Editor.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kravets-levko) [reply client/app/visualizations/gauge/Editor.jsx]: Replace 
```jsx
<input className="form-control" type="text" value={options.okRange} onChange={event => updateOptions({ okRange: event.target.value })} />
```
with
```jsx
import Input from 'antd/lib/input';

<Input value={options.okRange} onChange={event => updateOptions({ okRange: event.target.value })} />
```
The same for other text inputs.

For color inputs I'll prepare an example a little bit later.

**Comment by arikfr** [comment]
Seems like this is abandoned. I'm closing this now, but you're welcome to re-open after addressing the comments. Thanks!

**Pull request closed without merging**